src/main/java/com/google/cloud/spanner/pgadapter/utils/MutationWriter.java

```diff
@@ -134,27 +131,39 @@ public void buildMutation(ConnectionHandler connectionHandler, byte[] payload) t
                   + recordValue
                   + "\"");
         } catch (IllegalArgumentException e) {
-          rollback(connectionHandler, payload);
+          rollback(connectionHandler);
           throw new SQLException("Invalid input syntax for column \"" + columnName + "\"");
         } catch (Exception e) {
-          rollback(connectionHandler, payload);
+          rollback(connectionHandler);
           throw e;
         }
       }
       this.mutations.add(builder.build()); // Add write builder to mutation list
       this.mutationCount += record.size(); // Increment the number of mutations being added
       this.rowCount++; // Increment the number of COPY rows by one
     }
-    this.batchSize += payload.length; // Increment the batch size based on payload length
+    if (!mutationCountIsWithinLimit()) {
+      rollback(connectionHandler);
+      throw new SQLException(
+          "Mutation count: " + mutationCount + " has exceeded the limit: " + MUTATION_LIMIT);
+    }
+  }
+
+  /**
+   * @return True if current payload will fit within COMMIT_LIMIT. This is only an estimate and the
+   *     actual commit size may still be rejected by Spanner.
+   */
+  private boolean payloadFitsInCurrentBatch() {
```

Comment:
can we name this commitSizeIsWithinLimit to  match the naming of mutationCountIsWithinLimit? Also because the batching has been removed

Reply:
Done.